modules/luigi/__init__.py

```diff
@@ -2,90 +2,132 @@
+        lemmas = ctx[nlp.prop_lemmas]
+        if "flavor" in lemmas or "kind" in lemmas:
+            ctx[rawio.prop_out] = "i'm selling chocolate and vanilla today, both are pretty yummy... " \
+                                  "gonna be hard to choose for you"
+
+    @rs.state(
+        cond=nlp.prop_tokens.changed(),
+        read=(nlp.prop_tokens, nlp.prop_triples, nlp.prop_lemmas, nlp.prop_ner, prop_flavors, prop_scoops),
+        write=(prop_flavors, prop_scoops),
+        signal=sig_changed_flavor_or_scoops,
+        emit_detached=True)
+    def detect_flavors_and_scoops(ctx: rs.ContextWrapper):
         ice_cream_order = False
         tokens = ctx[nlp.prop_tokens]
         triples = ctx[nlp.prop_triples]
         lemmas = ctx[nlp.prop_lemmas]
-        if "chocolate" in tokens or "vanilla" in tokens:
-            if len(tokens) == 1:
-                # this case holds when customer orders ice creams using just the flavor name
-                ice_cream_order = True
-            elif triples[0].match_either_lemma(subj={"i"}) and \
-                    triples[0].match_either_lemma(pred={"want", "like", "desire", "have", "decide", "get",
-                                                        "choose", "wish", "prefer"}) and \
-                    triples[0].match_either_lemma(obj={"chocolate", "vanilla"}) and \
-                    "not" not in lemmas:
-                # this case holds when customer orders ice cream using phrases like
-                # "i would like to have chocolate please"
-                # "can i get some vanilla?"
-                # "i want the chocolate ice cream!"
-                # "i will have vanilla"
-                # "i have decided for chocolate"
-                # it does not hold whenever the customer negates his expression, i.e.
-                # "i don't like chocolate"
-                # "i have decided not to have vanilla"
-                ice_cream_order = True
-            elif triples[0].match_either_lemma(pred={"chocolate", "vanilla"}) and \
-                    "no" not in lemmas and "not" not in lemmas:
-                # this case holds when the order is phrased in a simple way like
-                # "chocolate please"
-                # "vanilla it is"
-                # in case of negation the order is not recognized
-                # "no chocolate ice cream please"
-                # "it is definitely not vanilla"
-                ice_cream_order = True
-            elif triples[0].match_either_lemma(subj={"chocolate", "vanilla"}) and \
-                    "no" not in lemmas and "not" not in lemmas:
-                # this case holds when the customer makes his order as follows:
-                # "vanilla sounds delicious"
-                # "chocolate would be good"
-                # again negations are ignored
-                ice_cream_order = True
+        ner = ctx[nlp.prop_ner]
+        if len(tokens) == 1 and "chocolate" in tokens or "vanilla" in tokens:
```

Comment:
Consider adding a constant set like `FLAVORS={...}` and change this line to `len(tokens & flavors) > 1`

Reply:
Actually (tokens & flavors) is enough - the empty set will evaluate to false, otherwise true 😺

Reply:
done